src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/CreateNewUserProfileWithIdamErrorsIntTest.java

```diff
@@ -98,6 +103,60 @@ void should_return_400_when_create_user_profile_has_invalid_role_and_response_is
                 .isEqualTo("13 Required parameters or one of request field is missing or invalid");
     }
 
+    @Test
+    void should_return_401_when_idam_server_throws_500_error() throws Exception {
+
+        UserProfileCreationData data = buildCreateUserProfileData();
+
+        setSidamRegistrationMockWithStatus(INTERNAL_SERVER_ERROR.value(), true);
+
+        ErrorResponse errorResponse =
+                userProfileRequestHandlerTest.sendPost(mockMvc, APP_BASE_PATH, data, UNAUTHORIZED, ErrorResponse.class);
+
+        assertThat(errorResponse.getErrorDescription()).isEqualTo("18 Unknown error from Idam");
```

Comment:
Where is the assert for 401 as mentioned in the test case title? (same for below tests)

Reply:
As current response doesnt returning Response status, Will have to modify the send response method and add required assert.I will work on it